src/chainstate/stacks/boot/mod.rs

```diff
@@ -260,206 +237,34 @@ impl StacksChainState {
-
-                        // if the last entry in `reward-cycle-pox-address-list` had an associated stacker,
-                        //   we must also update that stacker's `stacking-state`
-                        if let Some(stacker_val) = stacker {
-                            let moved_stacker = stacker_val.expect_principal();
-                            // load the `stacking-state`
-                            let moved_state_key = TupleData::from_data_static(vec![("stacker".into(), moved_stacker.clone().into())]).into();
-                            let mut moved_state_entry = db
-                                .expect_fetch_entry(
-                                    &pox_contract,
-                                    "stacking-state",
-                                    &moved_state_key,
-                                ).unwrap()
-                                .expect_tuple();
-                            // calculate the index into the reward-set-indexes list that
-                            //  this reward cycle is at
-                            let moved_cycle_index: usize = reward_cycle_to_update
-                                .checked_sub(
-                                    moved_state_entry
-                                        .get("first-reward-cycle")
-                                        .expect("Malformed tuple return by PoX contract")
-                                        .clone()
-                                        .expect_u128()
-                                )
-                                .expect("FATAL: Moved a reward set entry for a stacker whose first-reward-cycle was after the current cycle")
-                                .try_into()
-                                .expect("FATAL: list size is greater than usize");
-
-                            // update the list of reward set indexes
-                            let mut moved_reward_indexes = moved_state_entry
-                                .get("reward-set-indexes")
-                                .expect("Malformed tuple return by PoX contract")
-                                .clone()
-                                .expect_list();
-                            assert!(moved_reward_indexes.len() > moved_cycle_index, "FATAL: Calculated bad move index");
-                            // we moved the entry to the "target" location
-                            moved_reward_indexes[moved_cycle_index] = Value::UInt(target_entry_index);
-
-                            moved_state_entry.data_map.insert("reward-set-indexes".into(),
-                                                              Value::list_from(moved_reward_indexes)
-                                                                .expect("Failed to reconstruct Clarity list"));
-                            // store the new state back into the stacking-state map
-                            db.set_entry_unknown_descriptor(
+            let result = clarity
```

Comment:
Just want to confirm -- because this is an "internal" process, there's no block limit here, right?

Reply:
No -- I had to update this to explicitly switch to a free tracker.

Reply:
266caf2ba